### PR TITLE
Add uid_min and implied auid product.yml

### PR DESCRIPTION
**Pull request by yuumasato**
Avoid setting value of `auid` in every definition or fix. Lets set them product wise.

#### Description:

- Adds `uid_min` to product.yml.
  - If not defined, its default value is 1000
- `auid` is implied and set to `uid_min`.
- I have made them two separate values, because there are rules based on value of `uid_min` and there are the auditd rules that use `auid`, and it may be confusing to use them interchangeably.

#### Rationale:

- The `auid` parameter used in `auditd` rules should be set according to `uid_min` of the platform.

### CODE REVIEW

File: ssg/yaml.py [hunk 1]
```diff
@@ -46,6 +47,12 @@ def _get_implied_properties(existing_properties):
             "pkg_system" not in existing_properties):
         pkg_manager = existing_properties["pkg_manager"]
         result["pkg_system"] = PKG_MANAGER_TO_SYSTEM[pkg_manager]
+
+    if "uid_min" not in existing_properties:
+        result["uid_min"] = result["auid"] = DEFAULT_UID_MIN
+    Else:
```
Comment (mpreisler) [inline ssg/yaml.py]: `Else`?

Comment (yuumasato) [reply ssg/yaml.py]: /o\

File: ssg/yaml.py [hunk 2]
```diff
@@ -46,6 +47,12 @@ def _get_implied_properties(existing_properties):
             "pkg_system" not in existing_properties):
         pkg_manager = existing_properties["pkg_manager"]
         result["pkg_system"] = PKG_MANAGER_TO_SYSTEM[pkg_manager]
+
+    if "uid_min" not in existing_properties:
+        result["uid_min"] = result["auid"] = DEFAULT_UID_MIN
+    else:
+        result["auid"] = existing_properties["uid_min"]
```
Comment (mpreisler) [inline ssg/yaml.py]: This is doing two things, setting uid_min if it's not set in product and then setting auid according to uid_min. It doesn't respect any auid override so it's not clear to me why there even are 2 variables. Regardless, the result["auid"] logic should be a separate line. Handle uid_min, then handle auid, ...

Comment (yuumasato) [reply ssg/yaml.py]: I have updated the logic to not override `auid` if it is already set.

**Comment by jan-cerny** [comment]
I have checked the output OVALs for RHEL 6, 7 and Fedora.

IMO @mpreisler's comments have been adressed.

I think this can be merged safely.

Thank you for your contribution.

File: ssg/yaml.py [hunk 3]
```diff
@@ -46,6 +47,16 @@ def _get_implied_properties(existing_properties):
             "pkg_system" not in existing_properties):
         pkg_manager = existing_properties["pkg_manager"]
         result["pkg_system"] = PKG_MANAGER_TO_SYSTEM[pkg_manager]
+
+    if "uid_min" not in existing_properties:
+        result["uid_min"] = DEFAULT_UID_MIN
+
+    if "auid" not in existing_properties:
+        if "uid_min" in existing_properties:
+            result["auid"] = existing_properties["uid_min"]
```
Comment (mpreisler) [inline ssg/yaml.py]: there is a much nicer way to do this in Python, existing_properties.get("uid_min", DEFAULT_UID_MIN). It's much simpler and shorter

**Comment by mpreisler** [comment]
I wish the code was nicer but it's not a big deal. I added a comment anyway for future reference...

**Pull request merged**